ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetTransactionCount.java

```diff
@@ -45,14 +62,48 @@ public String getName() {
 
   @Override
   public JsonRpcResponse response(final JsonRpcRequest request) {
-    if (request.getParamLength() != 2) {
-      return new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
-    }
-
     final Address address = parameters.required(request.getParams(), 0, Address.class);
-    final String privacyGroupId = parameters.required(request.getParams(), 1, String.class);
+    final String privacyGroupId;
+    long nonce = 0;
+    try {
+      if (request.getParamLength() == 2) {
+        privacyGroupId = parameters.required(request.getParams(), 1, String.class);
+      } else if (request.getParamLength() == 3) {
+        final String privateFrom = parameters.required(request.getParams(), 1, String.class);
+        final String[] privateFor = parameters.required(request.getParams(), 2, String[].class);
+        privacyGroupId = determineLegacyPrivacyGroupId(privateFrom, privateFor);
+      } else {
+        return new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
+      }
```

Comment:
A cleaner code approach to consider might be using a switch statement in place of the `if`,`else if`, `else`
```suggestion
    switch(request.getParamLength()){
        case PRIVACY_GROUP_PARAM_LENGTH:
            privacyGroupId = parameters.required(request.getParams(), 1, String.class);
            break;

        case PRIVACY_GROUP_PARAM_LENGTH:
            final String privateFrom = parameters.required(request.getParams(), 1, String.class);
            final String[] privateFor = parameters.required(request.getParams(), 2, String[].class);
            privacyGroupId = determineLegacyPrivacyGroupId(privateFrom, privateFor);
            break;

        default:
            return new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
        }
    }
```

You'd also want to add these constants to the top of the file
`private static final int LEGACY_PRIVACY_PARAM_LENGTH = 3;`
`private static final int PRIVACY_GROUP_PARAM_LENGTH = 2;`